tasks/lint.rake

```diff
@@ -138,6 +159,11 @@ namespace :lint do
     FixmeLinter.new.run
   end
 
+  desc "Check that javascript bundle is up to date"
+  task :javascript_bundle do
+    JavascriptBundleLinter.new.run
+  end
+
```

Comment:
Thanks for adding the lint, I was thinking instead of doing something along the lines of the `spec/gemfiles_spec.lint.rb`. The reason is that, although it works, I think it's a bit artificial to use the `Linter` class here, for example, I don't like that calling `applicable_files` has the side effect of runninng `yarn build`, or that running the linter generates source control changes.

What about

#### spec/js_bundle_spec.lint.rb

```ruby
require "open3"

RSpec.describe "JS bundle sanity" do
  it "is up to date" do
    current_bundle = File.read("app/assets/javascripts/active_admin/base.js")

    output, status = Open3.capture2e("yarn build -o tmp/bundle.js")
    expect(status).to be_success, output

    new_bundle = File.read("tmp/bundle.js")

    msg = "Javascript bundle is out of date. Please run `yarn build` and commit changes."

    expect(current_bundle).to eq(new_bundle), msg
  end
end
```

Reply:
Makes complete sense to me indeed